### PR TITLE
Graph proactive bots and messages

### PR DESCRIPTION
**Pull request by laujan**

### CODE REVIEW

File: msteams-platform/graph-api/proactive-bots-&-messages/graph-proactive-bots-&-messages.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+|ad-hoc proactive message|dialog-based proactive message|
+|-----------------------------|------------------------------------|
+|The bot interjects a message without interrupting the conversation flow. | The bot creates a new dialog thread, takes control of a conversation, delivers the proactive message, closes, and returns control to the previous dialog.|
+
+## What is proactive app installation in Teams?
+
+Before your bot can proactively message a user, it needs to be installed either as a personal app, or in a team where the user is a member. At times,  you may need to proactively message users that have _not_ installed or previously interacted with your app. For instance, the need to message vital information to everyone in your organization. For that scenario you can use the Microsoft Graph API to proactively install your bot for your users and cache the necessary values received by your bot at install.
+
+Microsoft Graph [teamsAppInstallationresource type](/graph/api/resources/teamsappinstallation?view=graph-rest-1.0) permissions allow your bot to manage its installation lifecycle for all user (personal ) or team (group) scopes within the Microsoft Teams platform:
```
Comment (satadrubiswasmsft) [inline msteams-platform/graph-api/proactive-bots-&-messages/graph-proactive-bots-&-messages.md:27]: Broken link

File: msteams-platform/graph-api/proactive-bots-&-messages/graph-proactive-bots-&-messages.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+   "teamsApp@odata.bind" : "https://graph.microsoft.com/beta/appCatalogs/teamsApps/{teamsAppid}"
+}
+```
+
+### ✔ Retrieve conversation **chatId**
+
+**HTTP GET** request:
+
+```http
+ GET https://graph.microsoft.com/beta/users/{user-id}/teamwork/installedApps/{installation-id}/chat
```
Comment (satadrubiswasmsft) [inline msteams-platform/graph-api/proactive-bots-&-messages/graph-proactive-bots-&-messages.md:69]: Is `https://graph.microsoft.com/beta` intentional here? The others don't have `/beta` in the API URL

File: msteams-platform/graph-api/proactive-bots-&-messages/graph-proactive-bots-&-messages.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+```http
+GET /users/{user-id}/teamwork/installedApps?$expand=teamsAppDefinition&$filter=teamsAppDefinition/teamsAppId eq '{teamsAppid}'
+```
+
+### ✔ Install your app for subsequent users
+
+**HTTP POST** request:
+
+```http
+POST /users/{user-id}/teamwork/installedApps
```
Comment (satadrubiswasmsft) [inline msteams-platform/graph-api/proactive-bots-&-messages/graph-proactive-bots-&-messages.md:58]: A `/beta` will be a good idea since the v1.0 docs don't show the App installation for the user

File: msteams-platform/graph-api/proactive-bots-&-messages/graph-proactive-bots-&-messages.md [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+ > [!IMPORTANT]
+> Microsoft Graph will only install apps published in your organization's [app catalog](../../concepts/deploy-and-publish/overview.md#publish-to-your-organizations-app-catalog) or [AppSource](https://appsource.microsoft.com/).
+
+### ✔ Determine whether your bot is currently installed for all intended users
+
+**HTTP GET** request:
+
+```http
+GET /users/{user-id}/teamwork/installedApps?$expand=teamsAppDefinition&$filter=teamsAppDefinition/teamsAppId eq '{teamsAppid}'
```
Comment (satadrubiswasmsft) [inline msteams-platform/graph-api/proactive-bots-&-messages/graph-proactive-bots-&-messages.md:50]: A `/beta` will be a good idea since the v1.0 docs don't show the App installation for the user

File: msteams-platform/graph-api/proactive-bots-&-messages/graph-proactive-bots-&-messages.md (hunk 1 above)
Comment (laujan) [reply msteams-platform/graph-api/proactive-bots-&-messages/graph-proactive-bots-&-messages.md:27]: Relative file paths for URLs outside of our doc set only work on live site but they are checked during the build process.

File: msteams-platform/graph-api/proactive-bots-&-messages/graph-proactive-bots-&-messages.md [hunk 5]
```diff
@@ -0,0 +1,79 @@
+---
+title: Use Microsoft Graph to enable proactive bot installation and messaging in Teams
+description: Describes proactive messaging in Teams and how to implement.
+localization_priority:  Normal
+author: laujan
+ms.author: lajanuar
+ms.topic: Overview
+keywords: teams proactive messaging chat installation Graph
+---
+# Enable proactive bot installation and messaging in Teams with Microsoft Graph(Private Preview)
```
Comment (nkramer) [inline msteams-platform/graph-api/proactive-bots-&-messages/graph-proactive-bots-&-messages.md:10]: Space before the parenthesis.

I would explicitly use the phrase "proactive messaging" to help with SEO and recognition. So: "Enable proactive bot installation and **proactive** messaging in Teams with Microsoft Graph (Private Preview)".

Comment (nkramer) [reply msteams-platform/graph-api/proactive-bots-&-messages/graph-proactive-bots-&-messages.md:10]: Also, I like your phrase "proactive bot installation", wish I'd thought of it :)

File: msteams-platform/graph-api/proactive-bots-&-messages/graph-proactive-bots-&-messages.md [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+
+>[!NOTE]
+> Microsoft Graph private preview permissions may not achieve general availability status and currently should not be used in production.
+
+## What is proactive messaging in Teams?
+
+A proactive messages are initiated by bots to start conversations with a users. They serve many purposes including sending welcome messages, conducting surveys or polls, and sending organization-wide notifications.  Proactive messages in Teams can be delivered as either **ad-hoc** or **dialog-based** conversations:
```
Comment (nkramer) [inline msteams-platform/graph-api/proactive-bots-&-messages/graph-proactive-bots-&-messages.md:17]: The ad hoc vs dialog-based feels out of context in the intro. I've never had to think about this distinction when building bots, at least not in those terms, and I'm struggling to think of a practical example where you mix proactive messages with dialog-based flows.

File: msteams-platform/graph-api/proactive-bots-&-messages/graph-proactive-bots-&-messages.md [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+
+|ad-hoc proactive message|dialog-based proactive message|
+|-----------------------------|------------------------------------|
+|The bot interjects a message without interrupting the conversation flow. | The bot creates a new dialog thread, takes control of a conversation, delivers the proactive message, closes, and returns control to the previous dialog.|
+
+## What is proactive app installation in Teams?
+
+Before your bot can proactively message a user, it needs to be installed either as a personal app, or in a team where the user is a member. At times,  you may need to proactively message users that have _not_ installed or previously interacted with your app. For instance, the need to message vital information to everyone in your organization. For that scenario you can use the Microsoft Graph API to proactively install your bot for your users and cache the necessary values received by your bot at install.
```
Comment (nkramer) [inline msteams-platform/graph-api/proactive-bots-&-messages/graph-proactive-bots-&-messages.md:25]: I would leave out "and cache the necessary values received by your bot at install" as unnecessary to get the point across.

File: msteams-platform/graph-api/proactive-bots-&-messages/graph-proactive-bots-&-messages.md [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+> * Your bot must have _application permissions_ not _user delegated permissions_ because the installation is not for yourself but for others.
+>
+> * An Azure AD tenant administrator must [explicitly grant permissions to an application](/graph/security-authorization#grant-permissions-to-an-application). After an application is granted permissions, _all_ members of the Azure AD tenant will gain the granted permissions.
+
+## Enable proactive app installation and messaging
+
+ > [!IMPORTANT]
+> Microsoft Graph will only install apps published in your organization's [app catalog](../../concepts/deploy-and-publish/overview.md#publish-to-your-organizations-app-catalog) or [AppSource](https://appsource.microsoft.com/).
+
+### ✔ Determine whether your bot is currently installed for all intended users
```
Comment (nkramer) [inline msteams-platform/graph-api/proactive-bots-&-messages/graph-proactive-bots-&-messages.md:45]: Title says "all intended users" but you only show one user. I'd rephrase the title -- "is currently installed for the message recipient", maybe?

File: msteams-platform/graph-api/proactive-bots-&-messages/graph-proactive-bots-&-messages.md [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+
+**HTTP GET** request:
+
+```http
+GET /users/{user-id}/teamwork/installedApps?$expand=teamsAppDefinition&$filter=teamsAppDefinition/teamsAppId eq '{teamsAppid}'
+```
+
+### ✔ Install your app for subsequent users
```
Comment (nkramer) [inline msteams-platform/graph-api/proactive-bots-&-messages/graph-proactive-bots-&-messages.md:53]: It's not install for subsequent users, its install if app hasn't already been installed

File: msteams-platform/graph-api/proactive-bots-&-messages/graph-proactive-bots-&-messages.md [hunk 10]
```diff
@@ -0,0 +1,79 @@
+---
```
Comment (nkramer) [inline msteams-platform/graph-api/proactive-bots-&-messages/graph-proactive-bots-&-messages.md:1]: Is it a good idea to put a & in a filename?

File: msteams-platform/graph-api/proactive-bots-&-messages/graph-proactive-bots-&-messages.md [hunk 11]
```diff
@@ -0,0 +1,79 @@
...
+title: Use Microsoft Graph to enable proactive bot installation and messaging in Teams
```
Comment (nkramer) [inline msteams-platform/graph-api/proactive-bots-&-messages/graph-proactive-bots-&-messages.md:2]: How do you find this document? Is there a table of contents somewhere? How does this doc relate to https://docs.microsoft.com/en-us/microsoftteams/platform/bots/how-to/conversations/send-proactive-messages?tabs=dotnet ?

File: msteams-platform/graph-api/proactive-bots-&-messages/graph-proactive-bots-&-messages.md (hunk 2 above)
Comment (nkramer) [reply msteams-platform/graph-api/proactive-bots-&-messages/graph-proactive-bots-&-messages.md:69]: Need to tell people what to do with the results from the GET. ie, the id property of the result is the chatThread id you give to Bot Framework.

File: msteams-platform/graph-api/proactive-bots-&-messages/graph-proactive-bots-&-messages.md [hunk 12]
```diff
@@ -0,0 +1,79 @@
...
+
+**HTTP GET** request:
+
+```http
+ GET https://graph.microsoft.com/beta/users/{user-id}/teamwork/installedApps/{installation-id}/chat
+```
+
+### ✔ Send proactive messages
+
+Once your bot has been added to for a user or team and has acquired the necessary user  information, it can [Send proactive messsages](/azure/bot-service/bot-builder-howto-proactive-message?view=azure-bot-service-4.0&tabs=csharp)
```
Comment (nkramer) [inline msteams-platform/graph-api/proactive-bots-&-messages/graph-proactive-bots-&-messages.md:74]: The linked page goes on for quite a while without ever quite telling you what code you need to write. This page is more useful:
https://docs.microsoft.com/en-us/microsoftteams/platform/bots/how-to/conversations/send-proactive-messages?tabs=dotnet

Even that, though, takes a while to get to the punchline -- feels like we need in-line the code into this page:
// Construct the message to post to conversation
Activity newActivity = new Activity()
{
    Text = "Hello",
    Type = ActivityTypes.Message,
    Conversation = new ConversationAccount
    {
        Id = response.Id  // from GET https://graph.microsoft.com/beta/users/{user-id}/teamwork/installedApps/{installation-id}/chat
    },
};

// Post the message to chat conversation with user
await client.Conversations.SendToConversationAsync(newActivity, response.Id);

(And no, I've not tested that code -- probably best to do so before publishing this :) )

File: msteams-platform/graph-api/proactive-bots-&-messages/graph-proactive-bots-&-messages.md (hunk 3 above)
Comment (nkramer) [inline msteams-platform/graph-api/proactive-bots-&-messages/graph-proactive-bots-&-messages.md:58]: For each of these calls, let's provide a link to the corresponding Graph reference page

File: msteams-platform/graph-api/proactive-bots-&-messages/graph-proactive-bots-&-messages.md (hunk 12 above)
Comment (nkramer) [reply msteams-platform/graph-api/proactive-bots-&-messages/graph-proactive-bots-&-messages.md:74]: I think it's important, though, to make this process as easy as it can be to learn -- we get a fair amount of feedback that people find proactive messaging/installation too hard, and a lot of this is the conceptual hurdle of "bots are hard". Telling them that they just need a bot with five lines of code in it can make a big difference. (And yeah, I will also admit that calling four different APIs when you really wanted one API is harder than it needs to be, but that's the API we have today)

File: msteams-platform/graph-api/proactive-bots-&-messages/graph-proactive-bots-&-messages.md (hunk 5 above)
Comment (raghukolluru) [inline msteams-platform/graph-api/proactive-bots-&-messages/graph-proactive-bots-&-messages.md:10]: This is not private preview. It is public preview

File: msteams-platform/graph-api/proactive-bots-&-messages/graph-proactive-bots-&-messages.md (hunk 6 above)
Comment (laujan) [reply msteams-platform/graph-api/proactive-bots-&-messages/graph-proactive-bots-&-messages.md:17]: Those terms are mentioned in most of the MS proactive messaging articles so I thought they would be relevant in this context.

File: msteams-platform/graph-api/proactive-bots-&-messages/graph-proactive-bots-&-messages.md (hunk 10 above)
Comment (laujan) [reply msteams-platform/graph-api/proactive-bots-&-messages/graph-proactive-bots-&-messages.md:1]: Probably not 😏

File: msteams-platform/graph-api/proactive-bots-&-messages/graph-proactive-bots-&-messages.md (hunk 11 above)
Comment (laujan) [reply msteams-platform/graph-api/proactive-bots-&-messages/graph-proactive-bots-&-messages.md:2]: Updated the TOC; the above doc is a guide for v3 Bot Framework SDK.

File: msteams-platform/graph-api/proactive-bots-and-messages/graph-proactive-bots-and-messages.md [hunk 13]
```diff
@@ -0,0 +1,231 @@
...
+> Microsoft Graph will only install apps published in your organization's [app catalog](../../concepts/deploy-and-publish/overview.md#publish-to-your-organizations-app-catalog) or [AppSource](https://appsource.microsoft.com/).
+
+### ✔ [Create](../../bots/how-to/create-a-bot-for-teams.md) and [Publish](../../concepts/deploy-and-publish/overview.md) your [proactive messaging bot](../../concepts/bots/bot-conversations/bots-conv-proactive.md) for Teams
+
+When creating your app, make sure that you note the `id`  you use in your application manifest; you'll need it for the installation process.
+
+>[!TIP]
+> The production-ready [**Company Communicator**](../..//samples/app-templates.md#company-communicator) app template enables broadcast messaging and is a good foundation for building your proactive bot application.
+
+### ✔ Determine whether your bot is currently installed for a message recipient
```
Comment (AkJo) [inline msteams-platform/graph-api/proactive-bots-and-messages/graph-proactive-bots-and-messages.md:65]: We are missing the section on: how to go from the manifest id to teamsAppId. I explained that in our chat.

File: msteams-platform/graph-api/proactive-bots-and-messages/graph-proactive-bots-and-messages.md [hunk 14]
```diff
@@ -0,0 +1,256 @@
...
+
+**Microsoft Graph page reference.** [teamsApp resource type](/graph/api/resources/teamsapp?view=graph-rest-1.0)
+
+**HTTP GET** request:
+
+```http
+GET /appCatalogs/teamsApps?$filter=externalId eq '{IdFromManifest}'
+```
+
+The request will return a `teamsCatalogApp`  object. The returned object's `id`  is the app's catalog generated app ID and is different from the id that you provided in your Teams app manifest:
```
Comment (AkJo) [inline msteams-platform/graph-api/proactive-bots-and-messages/graph-proactive-bots-and-messages.md:75]: The request will return a 'teamsApp' object. 

The teamsCatalogApp documentation is incorrect, and we should fix that too, maybe as a separate PR.

File: msteams-platform/graph-api/proactive-bots-and-messages/graph-proactive-bots-and-messages.md [hunk 15]
```diff
@@ -0,0 +1,256 @@
...
+```
+
+### ✔ Determine whether your bot is currently installed for a message recipient
+
+**Microsoft Graph page reference:** [List apps installed for user](/graph/api/user-list-teamsappinstallation?view=graph-rest-beta&tabs=http)
+
+**HTTP GET** request:
+
+```http
+GET https://graph.microsoft.com/beta/users/{user-id}/teamwork/installedApps?$expand=teamsAppDefinition&$filter=teamsAppDefinition/teamsAppId eq '{teamsAppId}'
```
Comment (AkJo) [inline msteams-platform/graph-api/proactive-bots-and-messages/graph-proactive-bots-and-messages.md:98]: This can be simplified too:

GET https://graph.microsoft.com/beta/users/{user-id}/teamwork/installedApps?$expand=teamsApp&$filter=teamsApp/id eq '{teamsAppId}'

File: msteams-platform/graph-api/proactive-bots-and-messages/graph-proactive-bots-and-messages.md [hunk 16]
```diff
@@ -0,0 +1,256 @@
...
+
+If the user has Microsoft Teams running, they may see the app install immediately. Alternatively, a restart may be necessary to see the installed app.
+
+### ✔ Retrieve the conversation **chatId**
+
+When your app is installed for the user, the bot will receive a `conversationUpdate` [event notification](../../resources/bot-v3/bots-notifications.md#team-member-or-bot-addition) that will contain the necessary information to send the proactive message.
+
+The `chatId` can also be retrieved as follows:
+
+**Microsoft Graph reference:** [Get chat](/graph/api/chat-get?view=graph-rest-beta&tabs=http)
```
Comment (AkJo) [inline msteams-platform/graph-api/proactive-bots-and-messages/graph-proactive-bots-and-messages.md:124]: Instead of this which requires a different permission set, we should call out the new API to get the chat id associated with a bot:

GET https://graph.microsoft.com/beta/users/{user-id}/teamwork/installedApps/{installation-id}/chat 

The {installation-id} will be fetched from the response of 

GET https://graph.microsoft.com/beta/users/{user-id}/teamwork/installedApps?$expand=teamsApp&$filter=teamsApp/id eq '{teamsAppId}' 

We can then add: 
Alternatively, you can also get the chat id from:
GET https://graph.microsoft.com/beta/users/{user-id}/chats?$filter=installedApps/any(a:a/teamsApp/id eq '{teamsAppId}') ; however this needs the permission Chat.Read.All.

File: msteams-platform/graph-api/proactive-bots-and-messages/graph-proactive-bots-and-messages.md [hunk 17]
```diff
@@ -0,0 +1,256 @@
...
+      "id": "b1c5353a-7aca-41b3-830f-27d5218fe0e5",
+      "externalId": "f31b1263-ba99-435a-a679-911d24850d7c",
+      "name": "Test App",
+      "version": "1.0.1",
+      "distributionMethod": "Organization"
+    }
+  ]
+}
+```
+
```
Comment (AkJo) [inline msteams-platform/graph-api/proactive-bots-and-messages/graph-proactive-bots-and-messages.md:90]: Over here we can add something on the lines:

To fetch the teamsApp id of a sideloaded app, you would look in the collection that the sideloaded app is already installed to. Example:

GET users/{user-id}/teamwork/installedApps?$expand=teamsApp&$filter=teamsApp/externalId eq '{manifestId}'

GET teams/{team-id}/installedApps?$expand=teamsApp&$filter=teamsApp/externalId eq '{manifestId}'

File: msteams-platform/graph-api/proactive-bots-and-messages/graph-proactive-bots-and-messages.md [hunk 18]
```diff
@@ -0,0 +1,308 @@
...
+**HTTP GET** request:
+
+```http
+GET https://graph.microsoft.com/beta/users/{user-id}/teamwork/installedApps?$expand=teamsApp&$filter=teamsApp/id eq '{teamsAppId}'
+```
+
+2. Make the following request to fetch the `chatId`:
+
+**HTTP GET** request (permission — `Chat.Read`):
```
Comment (AkJo) [inline msteams-platform/graph-api/proactive-bots-and-messages/graph-proactive-bots-and-messages.md:163]: The permission for this API is TeamsAppInstallation.ReadWriteSelfForUser.All

File: msteams-platform/graph-api/proactive-bots-and-messages/graph-proactive-bots-and-messages.md [hunk 19]
```diff
@@ -0,0 +1,308 @@
...
+
+*See*, [Send proactive notifications to users SDK v4](/azure/bot-service/bot-builder-howto-proactive-message?view=azure-bot-service-4.0&tabs=csharp)
+
+## Proactive app installation in Teams
+
+Before your bot can proactively message a user, it needs to be installed either as a personal app, or in a team where the user is a member. At times,  you may need to proactively message users that have _not_ installed or previously interacted with your app. For instance, the need to message vital information to everyone in your organization. For such scenarios, you can use the Microsoft Graph API to proactively install your bot for your users.
+
+## Permissions
+
+Microsoft Graph [teamsAppInstallation resource type](/graph/api/resources/teamsappinstallation?view=graph-rest-1.0) permissions allow you to manage your app's installation lifecycle for all user (personal ) or team (group) scopes within the Microsoft Teams platform:
```
Comment (AkJo) [inline msteams-platform/graph-api/proactive-bots-and-messages/graph-proactive-bots-and-messages.md:32]: Space

Comment (AkJo) [inline msteams-platform/graph-api/proactive-bots-and-messages/graph-proactive-bots-and-messages.md:32]: I would avoid using the term 'group' because it will conflate with a another concent 'group chats'.

File: msteams-platform/graph-api/proactive-bots-and-messages/graph-proactive-bots-and-messages.md [hunk 20]
```diff
@@ -0,0 +1,308 @@
...
+
+**Microsoft Graph page reference:** [List apps installed for user](/graph/api/user-list-teamsappinstallation?view=graph-rest-beta&tabs=http)
+
+**HTTP GET** request:
+
+```http
+GET https://graph.microsoft.com/beta/users/{user-id}/teamwork/installedApps?$expand=teamsApp&$filter=teamsApp/id eq '{teamsAppId}'
+```
+
+3. If your app has already been uploaded/sideloaded for a channel in the group scope, you can retrieve the `teamsAppId` as follows:
```
Comment (AkJo) [inline msteams-platform/graph-api/proactive-bots-and-messages/graph-proactive-bots-and-messages.md:105]: team scope

File: msteams-platform/graph-api/proactive-bots-and-messages/graph-proactive-bots-and-messages.md [hunk 21]
```diff
@@ -0,0 +1,308 @@
...
+**Microsoft Graph reference:** [Get chat](/graph/api/chat-get?view=graph-rest-beta&tabs=http)
+
+1. You will need your app's  `{installation-id}`. If you don't have it, use the following:
+
+**HTTP GET** request:
+
+```http
+GET https://graph.microsoft.com/beta/users/{user-id}/teamwork/installedApps?$expand=teamsApp&$filter=teamsApp/id eq '{teamsAppId}'
+```
+
```
Comment (AkJo) [inline msteams-platform/graph-api/proactive-bots-and-messages/graph-proactive-bots-and-messages.md:160]: The id property of the response is the teamsAppInstallationId.

File: msteams-platform/graph-api/proactive-bots-and-messages/graph-proactive-bots-and-messages.md [hunk 22]
```diff
@@ -0,0 +1,308 @@
...
+### ✔ Retrieve the conversation **chatId**
+
+When your app is installed for the user, the bot will receive a `conversationUpdate` [event notification](../../resources/bot-v3/bots-notifications.md#team-member-or-bot-addition) that will contain the necessary information to send the proactive message.
+
+The `chatId` can also be retrieved as follows:
+
+**Microsoft Graph reference:** [Get chat](/graph/api/chat-get?view=graph-rest-beta&tabs=http)
+
+1. You will need your app's  `{installation-id}`. If you don't have it, use the following:
```
Comment (AkJo) [inline msteams-platform/graph-api/proactive-bots-and-messages/graph-proactive-bots-and-messages.md:153]: teamsAppInstallationId

File: msteams-platform/graph-api/proactive-bots-and-messages/graph-proactive-bots-and-messages.md [hunk 23]
```diff
@@ -0,0 +1,310 @@
...
+```
+
+2. If your app has already been uploaded/sideloaded for a user in the personal scope, you can retrieve the `teamsAppId` as follows:
+
+**Microsoft Graph page reference:** [List apps installed for user](/graph/api/user-list-teamsappinstallation?view=graph-rest-beta&tabs=http)
+
+**HTTP GET** request:
+
+```http
+GET https://graph.microsoft.com/beta/users/{user-id}/teamwork/installedApps?$expand=teamsApp&$filter=teamsApp/id eq '{teamsAppId}'
```
Comment (AkJo) [inline msteams-platform/graph-api/proactive-bots-and-messages/graph-proactive-bots-and-messages.md:102]: In this case, the user is trying to find the teamsAppId, so the query should be:

https://graph.microsoft.com/beta/users/{user-id}/teamwork/installedApps?$expand=teamsApp&$filter=teamsApp/externalId eq '{manifestId}'
```
